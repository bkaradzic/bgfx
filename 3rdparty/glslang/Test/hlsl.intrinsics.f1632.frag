float PixelShaderFunctionS(float inF0)
{
    f32tof16(inF0);

    return 0.0;
}

float1 PixelShaderFunction1(float1 inF0)
{
    // TODO: ... add when float1 prototypes are generated
    return 0.0;
}

float2 PixelShaderFunction2(float2 inF0)
{
    f32tof16(inF0);

    return float2(1,2);
}

float3 PixelShaderFunction3(float3 inF0)
{
    f32tof16(inF0);

    return float3(1,2,3);
}

float4 PixelShaderFunction(float4 inF0)
{
    f32tof16(inF0);

    return float4(1,2,3,4);
}

