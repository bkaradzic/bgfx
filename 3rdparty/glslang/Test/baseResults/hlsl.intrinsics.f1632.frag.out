hlsl.intrinsics.f1632.frag
ERROR: 0:3: 'f32tof16' : unimplemented intrinsic: handle natively 
ERROR: 0:16: 'f32tof16' : unimplemented intrinsic: handle natively 
ERROR: 0:23: 'f32tof16' : unimplemented intrinsic: handle natively 
ERROR: 0:30: 'f32tof16' : unimplemented intrinsic: handle natively 
ERROR: 4 compilation errors.  No code generated.


Shader version: 450
gl_FragCoord origin is upper left
ERROR: node is still EOpNull!
0:2  Function Definition: PixelShaderFunctionS(f1; (temp float)
0:2    Function Parameters: 
0:2      'inF0' (in float)
0:?     Sequence
0:3      ERROR: Bad unary op
 (temp uint)
0:3        'inF0' (in float)
0:5      Branch: Return with expression
0:5        Constant:
0:5          0.000000
0:9  Function Definition: PixelShaderFunction1(vf1; (temp 1-component vector of float)
0:9    Function Parameters: 
0:9      'inF0' (in 1-component vector of float)
0:?     Sequence
0:11      Branch: Return with expression
0:11        Constant:
0:11          0.000000
0:15  Function Definition: PixelShaderFunction2(vf2; (temp 2-component vector of float)
0:15    Function Parameters: 
0:15      'inF0' (in 2-component vector of float)
0:?     Sequence
0:16      ERROR: Bad unary op
 (temp 2-component vector of uint)
0:16        'inF0' (in 2-component vector of float)
0:18      Branch: Return with expression
0:?         Constant:
0:?           1.000000
0:?           2.000000
0:22  Function Definition: PixelShaderFunction3(vf3; (temp 3-component vector of float)
0:22    Function Parameters: 
0:22      'inF0' (in 3-component vector of float)
0:?     Sequence
0:23      ERROR: Bad unary op
 (temp 3-component vector of uint)
0:23        'inF0' (in 3-component vector of float)
0:25      Branch: Return with expression
0:?         Constant:
0:?           1.000000
0:?           2.000000
0:?           3.000000
0:29  Function Definition: PixelShaderFunction(vf4; (temp 4-component vector of float)
0:29    Function Parameters: 
0:29      'inF0' (layout(location=0 ) in 4-component vector of float)
0:?     Sequence
0:30      ERROR: Bad unary op
 (temp 4-component vector of uint)
0:30        'inF0' (layout(location=0 ) in 4-component vector of float)
0:32      Sequence
0:32        move second child to first child (temp 4-component vector of float)
0:?           '@entryPointOutput' (layout(location=0 ) out 4-component vector of float)
0:?           Constant:
0:?             1.000000
0:?             2.000000
0:?             3.000000
0:?             4.000000
0:32        Branch: Return
0:?   Linker Objects
0:?     '@entryPointOutput' (layout(location=0 ) out 4-component vector of float)
0:?     'inF0' (layout(location=0 ) in 4-component vector of float)


Linked fragment stage:


Shader version: 450
gl_FragCoord origin is upper left
ERROR: node is still EOpNull!
0:2  Function Definition: PixelShaderFunctionS(f1; (temp float)
0:2    Function Parameters: 
0:2      'inF0' (in float)
0:?     Sequence
0:3      ERROR: Bad unary op
 (temp uint)
0:3        'inF0' (in float)
0:5      Branch: Return with expression
0:5        Constant:
0:5          0.000000
0:9  Function Definition: PixelShaderFunction1(vf1; (temp 1-component vector of float)
0:9    Function Parameters: 
0:9      'inF0' (in 1-component vector of float)
0:?     Sequence
0:11      Branch: Return with expression
0:11        Constant:
0:11          0.000000
0:15  Function Definition: PixelShaderFunction2(vf2; (temp 2-component vector of float)
0:15    Function Parameters: 
0:15      'inF0' (in 2-component vector of float)
0:?     Sequence
0:16      ERROR: Bad unary op
 (temp 2-component vector of uint)
0:16        'inF0' (in 2-component vector of float)
0:18      Branch: Return with expression
0:?         Constant:
0:?           1.000000
0:?           2.000000
0:22  Function Definition: PixelShaderFunction3(vf3; (temp 3-component vector of float)
0:22    Function Parameters: 
0:22      'inF0' (in 3-component vector of float)
0:?     Sequence
0:23      ERROR: Bad unary op
 (temp 3-component vector of uint)
0:23        'inF0' (in 3-component vector of float)
0:25      Branch: Return with expression
0:?         Constant:
0:?           1.000000
0:?           2.000000
0:?           3.000000
0:29  Function Definition: PixelShaderFunction(vf4; (temp 4-component vector of float)
0:29    Function Parameters: 
0:29      'inF0' (layout(location=0 ) in 4-component vector of float)
0:?     Sequence
0:30      ERROR: Bad unary op
 (temp 4-component vector of uint)
0:30        'inF0' (layout(location=0 ) in 4-component vector of float)
0:32      Sequence
0:32        move second child to first child (temp 4-component vector of float)
0:?           '@entryPointOutput' (layout(location=0 ) out 4-component vector of float)
0:?           Constant:
0:?             1.000000
0:?             2.000000
0:?             3.000000
0:?             4.000000
0:32        Branch: Return
0:?   Linker Objects
0:?     '@entryPointOutput' (layout(location=0 ) out 4-component vector of float)
0:?     'inF0' (layout(location=0 ) in 4-component vector of float)

SPIR-V is not generated for failed compile or link
